server/src/main/java/io/crate/expression/symbol/LiteralValueFormatter.java

```diff
@@ -50,8 +50,11 @@ public void formatValue(Object value, StringBuilder builder) {
             formatIterable((Iterable<?>) value, builder);
         } else if (value.getClass().isArray()) {
             formatArray(value, builder);
-        } else if (value instanceof String || value instanceof Point || value instanceof Period) {
+        } else if (value instanceof String || value instanceof Point) {
             builder.append(Literals.quoteStringLiteral(value.toString()));
+        } else if (value instanceof Period) {
+            builder.append(Literals.quoteStringLiteral(value.toString()));
```

Comment:
This could also be handled as a conversion to an Interval Literal which would save one additional implicit cast.